src/main/kotlin/solve/scene/view/SceneView.kt

```diff
@@ -119,6 +112,32 @@ class SceneView : View() {
         controller.scrollY = null
     }
 
+    private fun addBindings() {
+        controller.sceneWidthProperty.bind(root.widthProperty())
+        controller.sceneProperty.onChange { scene ->
+            if (scene != null) {
+                frameDataLoadingScope.cancel()
+                frameDataLoadingScope = CoroutineScope(Dispatchers.Default)
+                draw()
+            }
+        }
+
+        // Grid settings bindings.
+        controller.columnsNumberProperty.onChange { columnsNumber ->
+            columnsNumber ?: return@onChange
+
+            currentGrid?.changeColumnsNumber(columnsNumber)
+        }
+        Platform.runLater {
+            currentWindow?.widthProperty()?.onChange {
+                Platform.runLater {
+                    controller.recalculateScale()
```

Comment:
Controller's scale calculations don't depend on the window size, even if it was so, you shouldn't scale frames when the window was resized.

Reply:
In this case we get a bit strange behavior if we don't want to have empty space around.
Without this code:
![without1](https://user-images.githubusercontent.com/66365072/232307881-b1c2be12-24dc-46b7-bb54-aff4f836d9f6.gif)

With it:
![without](https://user-images.githubusercontent.com/66365072/232307902-45250876-d43d-4467-94e9-0c4e3a89de41.gif)

Reply:
With this fix window resizing drops zooming, recalculate size only when empty space could appear or don't do anything
![JdB1YlvkLt](https://user-images.githubusercontent.com/55746901/232327221-971d22f0-8c37-4a18-aac4-e36f3c17d412.gif)